src/composables/auth.ts

```diff
@@ -277,20 +302,32 @@ export const useAuth = createCustomScopedComposable(() => {
     checkUserAuth();
   }
 
+  /**
+   * Check if password provided by user is correct and if true update it to new one.
+   */
+  async function updatePassword(currentPassword: string, newPassword: string) {
+    if (await authenticateWithPassword(currentPassword)) {
+      await setPassword(newPassword);
+    }
+  }
+
   (async () => {
     checkBiometricLoginAvailability();
 
-    const encryptedMnemonicExists = !!(WalletStorage.get<string>(STORAGE_KEYS.mnemonic));
-    if (
-      !encryptionKey.value
-        && !IS_MOBILE_APP
-        && encryptedMnemonicExists
-    ) {
-      await checkUserAuth();
+    await watchUntilTruthy(() => mnemonic.value);
+
+    if (!isMnemonicEncrypted.value) {
+      mnemonicDecrypted.value = mnemonic.value;
     }
 
-    if (IS_OFFSCREEN_TAB && !encryptionKey.value) {
-      await syncBackgroundEncryptionKey();
+    if ((IS_WEB || IS_EXTENSION) && !isMnemonicEncrypted.value) {
+      // Migrate the unprotected mnemonic by forcing user to set the password and encrypt it
+      await setMnemonicAndInitializePassword(mnemonic.value, true);
```

Comment:
This will not migrate the mnemonic, because user in this case would be asked to first enter a password.
![image](https://github.com/user-attachments/assets/008996a7-c03d-4543-bd8f-7c03e444e8ee)

Reply:
Nice catch. I updated my code. The migration now happens here: https://github.com/superhero-com/superhero-wallet/pull/3291/files#diff-dc9f684daabd265d4a2b08dfc8537c4f7925f66dae34b1859945a1fd889b3635R295